src/pkcs15init/pkcs15-openpgp.c

```diff
@@ -419,29 +419,38 @@ static int openpgp_store_data(struct sc_pkcs15_card *p15card, struct sc_profile
 	case SC_PKCS15_TYPE_CERT:
 		cinfo = (sc_pkcs15_cert_info_t *) obj->data;
 		cid = &(cinfo->id);
+		unsigned int tag = 0x7F21;
 
 		if (cid->len != 1) {
 			sc_log(card->ctx, "ID=%s is not valid.", sc_dump_hex(cid->value, cid->len));
 			LOG_FUNC_RETURN(card->ctx, SC_ERROR_INVALID_ARGUMENTS);
 		}
 
 		/* OpenPGP card v.2 contains only 1 certificate */
-		if (cid->value[0] != 3) {
+		if (cid->value[0] != 3 && p15card->card->type < SC_CARD_TYPE_OPENPGP_V3) {
 			sc_log(card->ctx,
 			       "This version does not support certificate ID = %d (only ID=3 is supported).",
 			       cid->value[0]);
 			LOG_FUNC_RETURN(card->ctx, SC_ERROR_NOT_SUPPORTED);
 		}
+
+
 		/* Just update the certificate DO */
+		u8 param = (u8) (2 - (cid->value[0] - 1));
+		sc_card_ctl(card, SC_CARDCTL_OPENPGP_SELECT_DATA, &param);
```

Comment:
please add error checking

Reply:
checking `sc_card_ctl`'s return value for `< 0` (also other occurrence in `libopensc/pkcs15-openpgp.c`)